Chess-Challenge/src/My Bot/MyBot.cs

```diff
@@ -98,25 +98,62 @@ static bool FindPromotionMove(Board board, IReadOnlyList<Move> moves, out Move p
     #endregion
 
     #region Capture
-    static bool FindCaptureMove(Board board, IReadOnlyList<Move> moves, out Move captureMove)
+    bool FindCaptureMove(Board board, IReadOnlyList<Move> moves, out Move captureMove)
+    {
+        captureMove = TryGetCaptureMove(board, moves, AfraidOfLosing);
+        return captureMove.IsCapture;
+    }
+
+    static Move TryGetCaptureMove(Board board, IReadOnlyList<Move> moves, bool lookForTwoMovesCapture = true, bool allowSacrifice = true)
     {
         Move resultMove = default;
-        void SaveMoveIfBest(Move move, bool preferable = false)
+        void SaveMoveIfBest(Move move, PieceType capturedPiece, bool preferable = false)
         {
-            if (move.CapturePieceType < resultMove.CapturePieceType) return;
-            if (!preferable && move.CapturePieceType == resultMove.CapturePieceType) return;
+            if (capturedPiece < resultMove.CapturePieceType) return;
+            if (!preferable && capturedPiece == resultMove.CapturePieceType) return;
             resultMove = move;
         }
         foreach (var move in moves)
         {
             if (move.IsCapture)
             {
-                if (move.MovePieceType < move.CapturePieceType) SaveMoveIfBest(move);
-                if (!MovedPeaceMayBeEaten(move, board)) SaveMoveIfBest(move, true);
+                if (allowSacrifice && move.MovePieceType < move.CapturePieceType) SaveMoveIfBest(move, move.CapturePieceType);
+                if (!MovedPeaceMayBeEaten(move, board)) SaveMoveIfBest(move, move.CapturePieceType, true);
             }
+            else if (lookForTwoMovesCapture)
+            {
+                // Try to capture in two moves
+                var capturedInTwoMoves = MakeMoveAndDoFunc(FindBestCaptureAfterOpponentMove, board, move);
+                if (capturedInTwoMoves > 0) SaveMoveIfBest(move, capturedInTwoMoves);
+            }
+        }
+        // TODO 4: Think of case when moved piece was a protector.
```

Comment:
Funny enough, this is the reason why it is hard to implement `cheapest enemy capture`. In fact, the cheapest move seems to be preferable by the enemy (the smart one). Naive `cheapest enemy capture` implementation worsens the results!